runtime/src/cost_model.rs

```diff
@@ -142,10 +142,14 @@ impl CostModel {
 
         // calculate bpf cost based on compute budget instructions
         let mut budget = ComputeBudget::default();
+        // for the purpose of getting requested compute unit for cost model, it is safe to enable
+        // request_heap_frame_ix regardless feature gate status
```

Comment:
```suggestion
        // request_heap_frame_ix regardless of its' feature gate status
```

Also, as someone not as familiar with this area of the code, why is this assumption safe?

Reply:
+1 deep lore and intuition should be explicitly documented.  don't make the next dev learn 10kLOC just to understand a comment

Reply:
it would also be good to assert this assumption with a test

Reply:
cost model currently only used by leaders when packing blocks. v1.13 is not using requested CU from compute_budget to determine transaction BPF costs (instead, it uses _estimated_ BPF instruction costs). So v1.13 leaders will always measure transactions' costs against block limits differently than v1.14+, but it'd never break consensus. Making v1.14+ to call `process_instruction` in logically correct way makes it easier to remove activated feature code later.

Reply:
Updated comments to include these info. Only v1.14 and above calls `compute_budget.process_instructions()`, v1.13 is on estimated instruction cost. So it is safe for v1.14+ to call CB in logically correct way.